contracts/zaps/OlympusZapV2.sol

```diff
@@ -0,0 +1,293 @@
+
+        require(tokensRec >= minToTokens, "High Slippage");
+
+        uint256 totalGoodwillPortion;
+
+        if (toToken == address(0)) {
+            totalGoodwillPortion = _subtractGoodwill(ETHAddress, tokensRec, affiliate, true);
+            payable(msg.sender).transfer(tokensRec - totalGoodwillPortion);
+        } else {
+            totalGoodwillPortion = _subtractGoodwill(toToken, tokensRec, affiliate, true);
+            IERC20(toToken).safeTransfer(msg.sender, tokensRec - totalGoodwillPortion);
+        }
+        tokensRec = tokensRec - totalGoodwillPortion;
+        emit zapOut(msg.sender, toToken, tokensRec, affiliate);
+    }
+
+    ////////////////////////// INTERNAL //////////////////////////
+
+    function _enterOlympus(uint256 amount, address toToken) internal returns (uint256) {
+        if (toToken == gOHM) {
+            IStaking(staking).stake(address(this), amount, false, false);
+            IStaking(staking).claim(address(this), false);
+            uint256 gOHMRec = IStaking(staking).wrap(msg.sender, amount);
+            return gOHMRec;
+        }
+        IStaking(staking).stake(msg.sender, amount, false, false);
+        IStaking(staking).claim(msg.sender, false);
+        return amount;
+    }
+
+    function _exitOlympus(address fromToken, uint256 amount) internal returns (uint256) {
+        if (fromToken == gOHM) {
+            uint256 sOHMRec = IStaking(staking).unwrap(address(this), amount);
+            IStaking(staking).unstake(msg.sender, sOHMRec, false, false);
+            return sOHMRec;
+        }
+        IStaking(staking).unstake(msg.sender, amount, false, false);
+        return amount;
+    }
+
+    function removeLiquidityReturn(address fromToken, uint256 fromAmount)
+        external
+        view
+        returns (uint256 ohmAmount)
+    {
+        if (fromToken == sOHM) {
+            return fromAmount;
+        } else if (fromToken == gOHM) {
+            return IsOHM(sOHM).fromG(fromAmount);
+        }
+    }
+
+    ////////////////////////// OLYMPUS ONLY //////////////////////////
+
+    function update_OlympusDAO(address _olympusDAO) external onlyOlympusDAO {
+        olympusDAO = _olympusDAO;
+    }
+
+    /// @notice update state for staking
+    function update_Staking(address _staking) external onlyOlympusDAO {
+        staking = _staking;
+    }
+
+    /// @notice update state for depo
+    function update_Depo(address _depo) external onlyOlympusDAO {
+        depo = _depo;
+    }
+
+    /// @notice update state for OHM
+    function update_OHM(address _OHM) external onlyOlympusDAO {
+        OHM = _OHM;
+    }
+
+    /// @notice update state for sOHM
+    function update_sOHM(address _sOHM) external onlyOlympusDAO {
+        sOHM = _sOHM;
+    }
+
+    /// @notice update state for gOHM
+    function update_gOHM(address _gOHM) external onlyOlympusDAO {
+        gOHM = _gOHM;
+    }
+
+    /// @notice update state for gOHM
+    function update_approvals(
```

Comment:
What is this used for?

Reply:
Its used for approving the depo for new principals, I also added an _approvalCheck() before any time transferFrom is called from the zap to allow the zap to function in the event that an approval is too low.

NOTE: the comment description has also been fixed.